pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -96,11 +97,31 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
-	count := 0
-	if m.hasParams() {
-		count = 1
-		for _, param := range m.Params {
-			count *= len(param.Value.ArrayVal)
+	// Explicit combinations in the Matrix
+	// Calculate based on length of include Names and return count
+	if m.hasInclude() && !m.hasParams() {
+		return len(m.Include)
+	}
+	paramNames := make(map[string][]string)
+	count := 1
+	// Iterate over matrix.params and compute count of all generated combinations
+	for _, param := range m.Params {
+		count *= len(param.Value.ArrayVal)
+		paramNames[param.Name] = param.Value.ArrayVal
+	}
```

Comment:
you can group this into a separate function - countCombinationsFromParams - which is then called here